flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/metrics/KubernetesClientMetricsTest.java

```diff
@@ -456,33 +339,101 @@ public void testAPIServerIsDown() {
 
         var deployment = TestUtils.buildApplicationCluster();
         mockServer.shutdown();
-        assertEquals(
-                0,
-                listener.getCounter(listener.getMetricId(REQUEST_FAILED_COUNTER_ID))
-                        .get()
-                        .getCount());
-        assertEquals(
-                0.0,
-                listener.getMeter(listener.getMetricId(REQUEST_FAILED_METER_ID)).get().getRate());
-        Awaitility.await()
-                .atMost(1, TimeUnit.MINUTES)
+
+        assertCounterIsZero(listener, REQUEST_FAILED_COUNTER_ID);
+        assertRateIsZero(listener, REQUEST_FAILED_METER_ID);
+
+        await().atMost(20, TimeUnit.SECONDS)
                 .until(
                         () -> {
                             assertThrows(
                                     KubernetesClientException.class,
                                     () -> kubernetesClient.resource(deployment).createOrReplace());
-                            return listener.getCounter(
-                                                            listener.getMetricId(
-                                                                    REQUEST_FAILED_COUNTER_ID))
-                                                    .get()
-                                                    .getCount()
-                                            > 0
-                                    && listener.getMeter(
-                                                            listener.getMetricId(
-                                                                    REQUEST_FAILED_METER_ID))
-                                                    .get()
-                                                    .getRate()
-                                            > 0.01;
+                            assertCounterIsPositive(listener);
+                            assertPositiveRate(listener, REQUEST_FAILED_METER_ID);
+                            return true;
                         });
     }
+
+    private static void assertRateIsZero(TestingMetricListener listener, String meterId) {
+        assertPositiveRate(listener, meterId, meter -> assertThat(meter.getRate()).isEqualTo(0.0));
```

Comment:
Testing for equality to a float might be problematic (unless `isEqualTo` does something clever)?

Would it be safer to test for > and < some small range either side of zero?

Reply:
I was surprised when checking assertJ was doing a simple comparison, so have switched to `isCloseTo` in c8d9b62622df1f9959b086ad7573e3bbbe7fed93.